samples/WinUI3/App.xaml.cs

```diff
@@ -0,0 +1,105 @@
+using Firebase.Auth;
+using Firebase.Auth.Providers;
+using Firebase.Auth.Repository;
+using Firebase.Auth.UI;
+using Microsoft.UI.Dispatching;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.Windows.AppLifecycle;
+using System;
+using Windows.Globalization;
+using Windows.UI.Core;
+
+namespace Auth.WinUI3.Sample
+{
+    public partial class App : Application
+    {        
+        public App()
+        {
+            this.InitializeComponent();
+
+            // Force override culture & language
+            ApplicationLanguages.PrimaryLanguageOverride = "cs";
+
+            // Firebase UI initialization
+            FirebaseUI.Initialize(new FirebaseUIConfig
+            {
+                ApiKey = "<YOUR API KEY>",
+                AuthDomain = "<YOUR PROJECT DOMAIN>.firebaseapp.com",
+                Providers = new FirebaseAuthProvider[]
+                {
+                    new GoogleProvider(),
+                    new FacebookProvider(),
+                    new AppleProvider(),
+                    new TwitterProvider(),
+                    new GithubProvider(),
+                    new MicrosoftProvider(),
+                    new EmailProvider()
+                },
+                PrivacyPolicyUrl = "https://github.com/step-up-labs/firebase-authentication-dotnet",
+                TermsOfServiceUrl = "https://github.com/step-up-labs/firebase-database-dotnet",
+                IsAnonymousAllowed = true,
+                AutoUpgradeAnonymousUsers = true,
+                UserRepository = new StorageRepository(),
+                // Func called when upgrade of anonymous user fails because the user already exists
+                // You should grab any data created under your anonymous user, sign in with the pending credential
+                // and copy the existing data to the new user
+                // see details here: https://github.com/firebase/firebaseui-web#upgrading-anonymous-users
+                AnonymousUpgradeConflict = conflict => conflict.SignInWithPendingCredentialAsync(true)
+            });
+        }
+
+        private void AuthStateChanged(object sender, UserEventArgs e)
+        {
+            var dispatcherQueue = DispatcherQueue.GetForCurrentThread();
```

Comment:
This returns null for me sometimes, would it be better to use the `Window?.DispatcherQueue` as we need to run this code on the main thread?

Reply:
I don't have this problem but still I saved dispatcher queue from OnLaunched where it should never be null and in AuthStateChanged I use this variable. Could you please check the problem is solved?